core/src/main/scala/fs2/job/Event.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2014–2018 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.job
+
+import scala.{Product, Serializable}
+import scala.concurrent.duration.Duration
+
+import java.lang.Throwable
+import java.time.Instant
+
+sealed trait Event[I] extends Product with Serializable
+
+object Event {
+  final case class Completed[I](id: I, startingTime: Instant, duration: Duration) extends Event[I]
```

Comment:
Not sure if `Instant` and `Duration` are the best types here. `PreparationEvent` uses `OffsetDateTime` instead of `Instant`, but `Instant` makes it easier to work with milliseconds instead of seconds.

Reply:
Generally I use `Duration` and `FiniteDuration` for everything. I don't use `Instant` at all because its usage is so inconsistent with `Duration`'s.

Reply:
I replaced `Instant` with `OffsetDateTime`. So this should be ready for another look.